### PR TITLE
Add pages for routine details, create and update

### PR DESCRIPTION
**Pull request by Hubertsci**
Screenshot 2024-10-28 at 16 35 38
Screenshot 2024-10-28 at 16 36 04
Screenshot 2024-10-28 at 16 36 34

### CODE REVIEW

File: src/app/routines/routineCard/routineOptionsButton/RoutineOptionsButton.tsx [hunk 1]
```diff
@@ -1,14 +1,42 @@
...
+      queryClient.invalidateQueries({ queryKey: ['routines'] });
+    },
+  });
+
+  const handleClose = () => {
+    setOpenConfirmationDialog(false);
+  };
+
+  const handleConfirm = async () => {
+    try {
```
Comment (Valentine-456) [inline src/app/routines/routineCard/routineOptionsButton/RoutineOptionsButton.tsx:33]: I think that this error check may be redundant sincereact-query already provides error handling.
I suggest the following approach instead (what do you think):
```
const deleteRoutineMutation = useMutation(deleteRoutine, {
  onSuccess: () => {
    // Close the dialog on success
    setOpenConfirmationDialog(false);
  },
  onError: (error) => {
    // Handle the error, such as logging or showing an alert
    console.error("Error deleting routine:", error);
  },
});

...
const handleConfirm = () => {
  deleteRoutineMutation.mutate(routineId);
};
```

**Review by przemyslawren — APPROVED** [review]
lgtm

File: src/app/api/routineApi.ts [hunk 2]
```diff
@@ -56,3 +56,52 @@ export const fetchRoutines = async (query = ''): Promise<Routine[]> => {
 
   return filteredRoutines;
 };
+
+export const fetchRoutineDetails = async (routineId: string): Promise<Routine> => {
+  await new Promise((resolve) => setTimeout(resolve, 1000));
```
Comment (Valentine-456) [inline src/app/api/routineApi.ts:61]: I dont think we need these timeouts. Let our application be fast 😎

File: src/app/api/routineApi.ts [hunk 3]
```diff
@@ -56,3 +56,52 @@ export const fetchRoutines = async (query = ''): Promise<Routine[]> => {
...
+
+  const newRoutine: Routine = {
+    id: (Math.random()*1000).toString(),
+    name: data.name,
+    description: data.description,
+    likes: Math.floor(Math.random()*100000),
+    userLikes: false
+  };
+  routines.push(newRoutine);
+
```
Comment (Valentine-456) [inline src/app/api/routineApi.ts:85]: I understand that this is some intermediate solution. But I think we already can fetch and modify routines from the backend and database.
We already have all the necessary endpoints + we may find bugs when integrating frontend and backend

**Review by Valentine-456 — COMMENTED** [review]
This functionality looks good from a user perspective. Just minor changes are needed from in my opinion:

File: src/app/api/routineApi.ts (hunk 2 above)
Comment (Hubertsci) [reply src/app/api/routineApi.ts:61]: These are just mocked functions which should be replaced with real api calls. It's the scope of another task in jira.

File: src/app/api/routineApi.ts (hunk 3 above)
Comment (Hubertsci) [reply src/app/api/routineApi.ts:85]: yeah but as mentioned we already have this task in jira in to do

File: src/app/api/routineApi.ts (hunk 2 above)
Comment (Valentine-456) [reply src/app/api/routineApi.ts:61]: Yes, you are right, I see it

**Review by Valentine-456 — APPROVED** [review]
LGTM!

**Pull request merged**